Modules/_tkinter.c

```diff
@@ -3389,17 +3389,28 @@ DisableEventHook(void)
 #endif
 }
 
+static int
+module_clear(PyObject *mod)
+{
+    Py_CLEAR(Tkinter_TclError);
+    Py_CLEAR(Tkapp_Type);
+    Py_CLEAR(Tktt_Type);
+    Py_CLEAR(PyTclObject_Type);
+    return 0;
+}
+
+static void
+module_free(void *mod)
+{
+    module_clear((PyObject *)mod);
+}
 
 static struct PyModuleDef _tkintermodule = {
     PyModuleDef_HEAD_INIT,
-    "_tkinter",
-    NULL,
-    -1,
```

Comment:
You removed `.m_size = -1,` but it's ok, `m_size=0` also means "no module state".

Reply:
There is a difference between 0 and -1 related to subinterpreters.

Reply:
What is the difference?

Reply:
From the documentation:
```rst
      Setting ``m_size`` to ``-1`` means that the module does not support
      sub-interpreters, because it has global state.
```

Reply:
Oh I see, I forgot about this difference :-(